addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -100,6 +109,9 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
         if (result.new_amount_total_raw !== undefined) {
             this._updateShippingCost(result.new_amount_total_raw);
         }
+        if (!this._areSelectedPaymentAndCarrierCompatible()) {
+            $payButton.prop('disabled', true);
+        }
```

Comment:
Could you add a simple test that ensures the button is disabled for incompatible carriers ?

Reply:
I don't think that this test would be "simple". Adding a tour for this case only seems a bit overkill to me.